### ISSUE TITLE
`Password::broker()->createToken()` reported as `UndefinedInterfaceMethod`

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1231]
**Describe the bug**

`Password::broker(...)->createToken($user)` is reported as `UndefinedInterfaceMethod`.

The `Password` facade's `broker()` docblock declares the return type as the contract `\Illuminate\Contracts\Auth\PasswordBroker`, which only declares `sendResetLink()` and `reset()`. The methods `createToken()`, `deleteToken()`, `tokenExists()`, `getUser()`, `getRepository()` exist only on the concrete `\Illuminate\Auth\Passwords\PasswordBroker`, which is what `PasswordBrokerManager::broker()` actually returns.

Calling those methods statically on the facade works fine (the facade has its own `@method static` annotations), only the `broker(...)->...` path is affected.

```text
UndefinedInterfaceMethod: Method Illuminate\Contracts\Auth\PasswordBroker::createToken does not exist
```

**Code to reproduce**

```php
use Illuminate\Contracts\Auth\CanResetPassword;
use Illuminate\Support\Facades\Password;

function repro(CanResetPassword $user): string
{
    return Password::broker('users')->createToken($user); // UndefinedInterfaceMethod
}
```

**Impacted Versions**

```text
laravel/framework       v13.19.0
psalm/plugin-laravel    dev-master 6ac7789
vimeo/psalm             dev-master 290eaec
```

**Psalm config**

Reproduces with a minimal config, the issue is purely type-level:

```xml
<?xml version="1.0"?>
<psalm errorLevel="1">
    <projectFiles>
        <directory name="app"/>
    </projectFiles>
    <plugins>
        <pluginClass class="Psalm\LaravelPlugin\Plugin"/>
    </plugins>
</psalm>
```

**Additional context**

Widening the contract upstream in laravel/framework would be a BC break for custom broker implementations, so a plugin-side stub seems like the right place: override `PasswordBrokerManager::broker()` (and the facade's `broker()` mapping) to return `\Illuminate\Auth\Passwords\PasswordBroker`. Similar in spirit to how `FacadeMapProvider` already special-cases `Auth` guard methods exposed via `AuthManager::createTokenDriver()`.

**Comment by alies-dev** [comment]
Resolved by [PR #1240](https://github.com/psalm/psalm-plugin-laravel/pull/1240), which adds provenance-based narrowing for `PasswordBrokerManager::broker()` / `Password::broker()` to the concrete `Illuminate\Auth\Passwords\PasswordBroker`. This fixes `createToken()` and the other concrete broker methods while leaving bare contract-typed values unchanged. Closing as completed.
